src/Tutor.Core/DomainModel/KnowledgeComponents/KnowledgeComponentMastery.cs

```diff
@@ -10,13 +10,19 @@ public class KnowledgeComponentMastery : EventSourcedAggregateRoot
         public double Mastery { get; private set; }
         public KnowledgeComponent KnowledgeComponent { get; private set; }
         public int LearnerId { get; private set; }
+        public bool IsPassed { get; private set; }
+        public bool IsCompleted { get; private set; }
+        public bool IsSatisfied { get; private set; }
 
         private KnowledgeComponentMastery() { }
 
         public KnowledgeComponentMastery(KnowledgeComponent knowledgeComponent)
         {
             Mastery = 0.0;
             KnowledgeComponent = knowledgeComponent;
+            IsPassed = false;
```

Comment:
I'm still quite unsure of this decision. What is the justification for tracking each field?

Reply:
Refactored to not store IsCompleted in [cd418f5](https://github.com/Clean-CaDET/tutor/pull/24/commits/cd418f5fd84c3d8b8dcd5bc6b935154c35919621).